reactor-core/src/main/java/reactor/core/publisher/FluxOnAssembly.java

```diff
@@ -434,6 +436,27 @@ final Throwable fail(Throwable t) {
 				}
 
 				t = Exceptions.addSuppressed(t, onAssemblyException);
+				final StackTraceElement[] stackTrace = t.getStackTrace();
+				if (stackTrace.length > 0) {
+					StackTraceElement[] newStackTrace = new StackTraceElement[stackTrace.length];
```

Comment:
should we treat an array of size one as a special case? that might be overoptimization because I don't see this depth of stacktrace as something that could happen very often, if at all, though... 😅

Reply:
to keep the code simple, I would not consider this a real world case, because there always will be at least two Reactor frames :)